target/linux/ipq806x/base-files/lib/upgrade/asrock.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+
+. /lib/functions.sh
+
+asrock_upgrade_prepare() {
+	local mtdnum="$( find_mtd_index 0:BOOTCONFIG )"
+	local ubidev="$( nand_find_ubi ubi )"
+
+	if [ ! "$mtdnum" ]; then
+		echo "cannot find bootconfig mtd partition"
+		return 1
+	fi
+
+
+
+	# Just delete these partitions if present and use
+	# OpenWrt's standard names for those.
+	ubirmvol /dev/$ubidev -N ubi_rootfs &> /dev/null || true
+	ubirmvol /dev/$ubidev -N ubi_rootfs_data &> /dev/null || true
+
+
+	#Set upgrade flag. If something goes wrong, router will boot with
+	#factory firmware.
+
+	local partition_byte=`dd if=/dev/mtd$mtdnum bs=1 skip=52 count=1`
+
+        dd if=/dev/mtd$mtdnum of=/tmp/mtd$mtdnum bs=1k
+        printf '\x01' | dd of=/tmp/mtd$mtdnum conv=notrunc bs=1 seek=4
+        printf '\x01' | dd of=/tmp/mtd$mtdnum conv=notrunc bs=1 seek=56
+
+        if [ -s $partition_byte ]
+        then
+                printf '\x01' | dd of=/tmp/mtd$mtdnum conv=notrunc bs=1 seek=52
+        else
+                printf '\x00' | dd of=/tmp/mtd$mtdnum conv=notrunc bs=1 seek=52
+        fi
+
+        mtd write /tmp/mtd$mtdnum /dev/mtd$mtdnum
+
+}
```

Comment:
broken indent here; just use tabs-only. Empty line before closing bracket can be dropped.